src/Hl7.Fhir.Core/Rest/HttpToEntryExtensions.cs

```diff
@@ -151,6 +151,10 @@ private static Resource parseResource(string bodyText, string contentType, Parse
 
                 return null;
             }
+            catch (ArgumentException ae) when (!throwOnArgumentException)
```

Comment:
We should not do this - instead, the error raised (currently ArgumentException) should be turned into a FormatException so it is caught by the original catch handler.